pkg/cmd/pulumi/state/state_unprotect.go

```diff
@@ -106,6 +109,70 @@ func unprotectAllResources(ctx context.Context, ws pkgWorkspace.Context, stackNa
 	return nil
 }
 
+// unprotectMultipleResources unprotects multiple resources specified by their URNs.
+func unprotectMultipleResources(
+	ctx context.Context, ws pkgWorkspace.Context, stackName string, urns []string, showPrompt bool,
+) error {
+	opts := display.Options{
+		Color: cmdutil.GetGlobalColorization(),
+	}
+	s, err := cmdStack.RequireStack(
+		ctx,
+		ws,
+		backend.DefaultLoginManager,
+		stackName,
+		cmdStack.OfferNew,
+		opts,
+	)
+	if err != nil {
+		return err
+	}
+
+	return TotalStateEdit(ctx, s, showPrompt, opts, func(_ display.Options, snap *deploy.Snapshot) error {
+		if snap == nil {
+			return errors.New("no resources found to unprotect")
+		}
+
+		var errs []string
+		resourceCount := 0
+
+		for _, urnStr := range urns {
+			urn := resource.URN(urnStr)
+			found := false
+
+			for _, res := range snap.Resources {
+				if res.URN == urn {
+					found = true
+					if err := edit.UnprotectResource(snap, res); err != nil {
+						errs = append(errs, fmt.Sprintf("error unprotecting resource %q: %v", urn, err))
+					} else {
```

Comment:
You can use a `contract.Assert` here instead of the error checking.  `UnprotectResource` will never return an error. (This is a bad API that we need to fix)